launcher/ui/pages/instance/ExternalResourcesPage.cpp

```diff
@@ -289,6 +302,27 @@ void ExternalResourcesPage::disableItem()
     m_model->setResourceEnabled(selection.indexes(), EnableAction::DISABLE);
 }
 
+void ExternalResourcesPage::viewHomepage()
+{
+    auto selection = m_filterModel->mapSelectionToSource(ui->treeView->selectionModel()->selection()).indexes();
+    bool openedAny = false;
+    for (auto resource : m_model->selectedResources(selection)) {
+        auto url = resource->homepage();
+        if (!url.isEmpty()) {
+            DesktopServices::openUrl(url);
+            openedAny = true;
+        }
+    }
+
+    // TODO: just disable button
+    // just doing this for now to prevent race conditions which may be worse with implementation changes
+    if (!openedAny) {
+        CustomMessageBox::selectable(this, tr("No homepages found"), tr("None of the selected resources had an available homepage."),
+                                     QMessageBox::Warning, QMessageBox::Ok, QMessageBox::Ok)
+            ->exec();
+    }
```

Comment:
I don't think we need to display a message here.

Reply:
If not, no pages open and the user isn't aware. I changed it not to disable the button as if you click on a mod before data is loaded then you can't click it after it's loaded (this will be more likely if we switch to lazy loading which I think we probably will)

Reply:
I do feel like this code should not be called: so better fix the button disable.

Reply:
Will do, just note it might make the button unpressable with lazy loading if we decide to do that but it can be addressed then a suppose